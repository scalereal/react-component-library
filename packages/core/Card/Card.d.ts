import * as React from 'react';
import { WithStyle } from '../../utils';
import { CardProps } from './types';
export declare const variants: CardProps['variant'][];
export declare const textAlignments: CardProps['textAlignment'][];
export declare const textSizes: CardProps['titleSize'][];
export declare const descriptionSizes: CardProps['descriptionSize'][];
export declare const Card: React.FC<CardProps> & WithStyle;
export default Card;
