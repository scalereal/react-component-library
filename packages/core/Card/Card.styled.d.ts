import { CardProps } from './types';
export declare const Card: import("styled-components").StyledComponent<"div", any, CardProps, never>;
export declare const CardHeader: import("styled-components").StyledComponent<"div", any, CardProps, never>;
export declare const CardImage: import("styled-components").StyledComponent<"img", any, {}, never>;
export declare const CardTitle: import("styled-components").StyledComponent<"div", any, {}, never>;
export declare const CardBody: import("styled-components").StyledComponent<"div", any, CardProps, never>;
export declare const CardDescription: import("styled-components").StyledComponent<"div", any, {}, never>;
