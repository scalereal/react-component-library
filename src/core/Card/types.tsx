export type CardProps =  {
     /** Card Image */
    src:string;
    /** Card Title */
    title: string;
    /** Card message */
    message: string;
};
