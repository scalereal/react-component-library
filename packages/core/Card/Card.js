var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
import React from 'react';
import Text from '../Text/Text';
import * as Styled from './Card.styled';
export var variants = ['horizontal', 'vertical'];
export var textAlignments = ['left', 'center', 'right'];
export var textSizes = ['XS', 'S', 'M', 'L', 'XL'];
export var descriptionSizes = ['XS', 'S', 'M', 'L', 'XL'];
var Card = React.memo(React.forwardRef(function (props, ref) {
    var title = props.title, description = props.description, imgSrc = props.imgSrc, titleSize = props.titleSize, descriptionSize = props.descriptionSize, titleColor = props.titleColor, descriptionColor = props.descriptionColor;
    return (React.createElement(Styled.Card, __assign({ ref: ref }, props),
        imgSrc && React.createElement(Styled.CardHeader, __assign({}, props),
            React.createElement(Styled.CardImage, { src: imgSrc })),
        React.createElement(Styled.CardBody, __assign({}, props),
            React.createElement(Styled.CardTitle, null,
                React.createElement(Text, { textColor: titleColor, textSize: titleSize }, title)),
            React.createElement(Styled.CardDescription, null,
                React.createElement(Text, { textColor: descriptionColor, textSize: descriptionSize }, description)))));
}));
Card.defaultProps = {
    textAlignment: 'left',
    variant: 'vertical'
};
Card.displayName = 'Card';
Card.Style = Styled.Card;
export default Card;
