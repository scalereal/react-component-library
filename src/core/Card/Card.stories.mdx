import { Card, variants, textAlignments, textSizes, descriptionSizes } from './Card';
import { Preview, Story, Meta, Props, ArgsTable, Canvas } from '@storybook/addon-docs/blocks';
import { withKnobs, color, boolean, select } from '@storybook/addon-knobs';

<Meta
    title="Components/Core/Card"
    component={Card}
    argTypes={{
        shadowColor: {
            control: 'color'
        },
        borderColor: {
            control: 'color'
        },
        titleColor: {
            control: 'color'
        },
        descriptionColor: {
            control: 'color'
        },
        variant: {
            control: {
                type: 'select',
                options: Object.values(variants)
            }
        },
        textAlignment: {
            control: {
                type: 'select',
                options: Object.values(textAlignments)
            }
        },
        titleSize: {
            control: {
                type: 'select',
                options: Object.values(textSizes)
            }
        },
        descriptionSize: {
            control: {
                type: 'select',
                options: Object.values(descriptionSizes)
            }
        }
    }}
/>

# Card

Displays a Card

export const Template = ({
    imgSrc,
    title,
    description,
    variant,
    textAlignment,
    borderColor,
    shadowColor,
    titleSize,
    descriptionSize,
    ...args
}) => {
    return (
        <Card
            imgSrc={imgSrc}
            title={title}
            description={description}
            variant={variant}
            textAlignment={textAlignment}
            borderColor={borderColor}
            shadowColor={shadowColor}
            titleSize={titleSize}
            descriptionSize={descriptionSize}
            {...args}
        />
    );
};

<Preview>
    <Story
        name="Card"
        args={{
            title: 'Title',
            description: 'Lorem Ipsum is simply dummy text',
            imgSrc: 'https://www.w3schools.com/w3css/img_forest.jpg',
            variant: 'vertical',
            textAlignment: 'left',
            titleSize: 'M',
            descriptionSize: 'XS'
        }}
    >
        {Template.bind({})}
    </Story>
</Preview>

### Color Variations

You can either change the theme or pass the colors as props.

<Preview withToolbar>
    <div>
        <Card
            title="Title"
            description="Lorem Ipsum is simply dummy text"
            imgSrc="https://www.w3schools.com/w3css/img_forest.jpg"
            variant="vertical"
            textAlignment="left"
            titleSize="M"
            descriptionSize="XS"
            borderColor="blue"
            titleColor="green"
            descriptionColor="blue"
            shadowColor="blue"
        />
    </div>
</Preview>

### Variant

Displays card in vertical or horizontal

<Preview withToolbar>
    <div>
        <Card
            title="Title"
            description="Lorem Ipsum is simply dummy text"
            imgSrc="https://www.w3schools.com/w3css/img_forest.jpg"
            variant="vertical"
            textAlignment="left"
            titleSize="M"
            descriptionSize="XS"
            variant="horizontal"
        />
    </div>
</Preview>

### textAlignment

`lef` | `center` | `right`

<Preview withToolbar>
    <div style={{ display: 'flex', flexWrap: 'wrap', justifyContent: 'space-around' }}>
        <Card
            title="Title"
            description="Lorem Ipsum is simply dummy text"
            imgSrc="https://www.w3schools.com/w3css/img_forest.jpg"
            variant="vertical"
            textAlignment="left"
            titleSize="S"
            descriptionSize="XS"
        />
        <Card
            title="Title"
            description="Lorem Ipsum is simply dummy text"
            imgSrc="https://www.w3schools.com/w3css/img_forest.jpg"
            variant="vertical"
            textAlignment="center"
            titleSize="S"
            descriptionSize="XS"
        />
        <Card
            title="Title"
            description="Lorem Ipsum is simply dummy text"
            imgSrc="https://www.w3schools.com/w3css/img_forest.jpg"
            variant="vertical"
            textAlignment="right"
            titleSize="S"
            descriptionSize="XS"
        />
    </div>
</Preview>

### Text Sizes

There are five sizs available `XS` | `S` | `M` | `L` | `XL`.

<Preview withToolbar>
    <div style={{ display: 'flex', flexWrap: 'wrap', justifyContent: 'space-around' }}>
        <Card
            title="Title"
            description="Lorem Ipsum is simply dummy text"
            imgSrc="https://www.w3schools.com/w3css/img_forest.jpg"
            variant="vertical"
            textAlignment="left"
            titleSize="XS"
            descriptionSize="XS"
        />
        <Card
            title="Title"
            description="Lorem Ipsum is simply dummy text"
            imgSrc="https://www.w3schools.com/w3css/img_forest.jpg"
            variant="vertical"
            textAlignment="left"
            titleSize="S"
            descriptionSize="XS"
        />
        <Card
            title="Title"
            description="Lorem Ipsum is simply dummy text"
            imgSrc="https://www.w3schools.com/w3css/img_forest.jpg"
            variant="vertical"
            textAlignment="left"
            titleSize="M"
            descriptionSize="S"
        />
        <Card
            title="Title"
            description="Lorem Ipsum is simply dummy text"
            imgSrc="https://www.w3schools.com/w3css/img_forest.jpg"
            variant="vertical"
            textAlignment="left"
            titleSize="L"
            descriptionSize="M"
        />
        <Card
            title="Title"
            description="Lorem Ipsum is simply dummy text"
            imgSrc="https://www.w3schools.com/w3css/img_forest.jpg"
            variant="vertical"
            textAlignment="left"
            titleSize="XL"
            descriptionSize="L"
        />
    </div>
</Preview>

### Props

<Props story="Card" />
