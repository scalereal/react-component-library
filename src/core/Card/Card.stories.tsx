import { Meta, Story } from '@storybook/react/types-6-0';
import React from 'react';
import Card from "./Card"


export default {
    title: 'Components/Core/Card',
    component: Card,
} as Meta

const Template:Story = ({ img, title, message, ...args}) => {
    return <Card src={img} title={title} message={message} {...args} />
}

export const Default = Template.bind({});
Default.args = {
    title:'Title',
    message:'Lorem Ipsum is simply dummy text',
    src:"https://hips.hearstapps.com/hmg-prod.s3.amazonaws.com/images/close-up-of-tulips-blooming-in-field-royalty-free-image-1584131616.jpg?crop=0.630xw:1.00xh;0.186xw,0&resize=640:*"
}
