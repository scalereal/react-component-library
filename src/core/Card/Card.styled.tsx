
import styled from "styled-components"
import { CardProps } from './types';
import { defaultTheme } from "../../theme/theme"

export const Card = styled('div')<CardProps>`
    display: flex;
    flex-direction:column;
    border: 1px solid ${defaultTheme.shadowColor};
    border-radius: 3px;
    max-width:20rem;
    box-shadow: 0 0.5rem 2rem ${defaultTheme.shadowColor};
    width:16rem;
    height:100%;
`;
export const CardHeader = styled('div')`
    background-color:#fff;
`;
export const CardImage = styled('img')`
    width:100%;
    height:100%;
`;
export const CardTitle = styled('div')`
    font-size:1.2em;
    color: ${defaultTheme.typographyColor};
`;
export const CardBody = styled('div')`
    padding:0.9rem 0.8rem;
    background-color:#fff;
    height:100%;
`;
export const CardMessage = styled('div')`
    font-size:1em;
    color: ${defaultTheme.typographyColor};
    padding:5px 0;
`;

