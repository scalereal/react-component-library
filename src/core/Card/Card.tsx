import React,{ FC } from 'react'
import * as Styled from "./Card.styled"
import { CardProps } from './types';

const Toast:FC<CardProps> = ({ src, title, message } : CardProps) => {
    return (
        <Styled.Card>
            <Styled.CardHeader>
                <Styled.CardImage src={src}/>
            </Styled.CardHeader>
            <Styled.CardBody>
              <Styled.CardTitle>
                     {title}
              </Styled.CardTitle>
              <Styled.CardMessage>
                  {message}
              </Styled.CardMessage>
            </Styled.CardBody>
        </Styled.Card>
    )
}

export default Toast
