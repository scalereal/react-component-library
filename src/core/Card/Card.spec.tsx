import { render } from '@testing-library/react';
import * as React from 'react';
import Card from "./Card"


describe('Card', () => {
    it('should render title and message text', () => {
        const { container } = render(<Card src="Image" title="Title" message="Message" />);
        expect(container).toMatchSnapshot();
    });
});
