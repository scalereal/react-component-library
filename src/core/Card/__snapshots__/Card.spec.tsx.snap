// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render title and message text 1`] = `
<div>
  <div
    class="sc-bdfBwQ efpqlE"
  >
    <div
      class="sc-gsTCUz hzklTu"
    >
      <img
        class="sc-dlfnbm fAbRqY"
        src="Image"
      />
    </div>
    <div
      class="sc-eCssSg fYBSNH"
    >
      <div
        class="sc-hKgILt hkQNYV"
      >
        Title
      </div>
      <div
        class="sc-jSgupP ktxcFg"
      >
        Message
      </div>
    </div>
  </div>
</div>
`;
